app/models/concerns/article.rb

```diff
@@ -11,11 +13,11 @@ module Article
 
     before_validation :set_author, on: :create
     before_validation :set_user
-    before_validation :set_channel, on: :create, unless: -> { self.class_name == "ClaimDescription" }
+    before_validation :set_channel, on: :create, unless: -> { self.is_a?(ClaimDescription) }
     validates_presence_of :user
 
-    validates :channel, inclusion: { in: %w[imported manual api zapier] }, unless: -> { self.class_name == "ClaimDescription" }
-    enum channel: { imported: 0, manual: 1, api: 2, zapier: 3 }
+    validates :channel, inclusion: { in: %w[imported manual api zapier] }, unless: -> { self.is_a?(ClaimDescription) }
```

Comment:
@vasconsaurus please use `ARTICLE_CHANNELS.keys` instead of `%w[imported manual api zapier]` here.